__tests__/app.test.js

```diff
@@ -208,12 +217,89 @@ describe("/api/articles/:article_id/comments", () => {
       });
   });
 
-  test("Get: 200 sends back an empty array to the client, when article id exist, but there is no comments for the id", () => {
+  test("Post: 201 inserts a new comment to the comments table and returns the created comment to the client", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "always happy",
+    };
     return request(app)
-      .get("/api/articles/2/comments")
-      .expect(200)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
       .then((response) => {
-        expect(response.body.comments).toEqual([]);
+        expect(response.body.newComment).toMatchObject({
+          comment_id: 19,
+          body: "always happy",
+          article_id: 1,
+          author: "rogersop",
+          votes: 0,
+          created_at: expect.any(String),
+        });
+      });
+  });
+
+  test("Post: 400 sends an appropriate status and error message when field is missing", () => {
+    const newComment = {
+      username: "rogersop",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe(
+          "body and username are required fields."
+        );
+      });
+  });
+
+  test("Post: 201 inserts a new comment to the comments table and returns the created comment to the client, when extra field is provided", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "always happy",
+      user: "me",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        expect(response.body.newComment).toMatchObject({
+          comment_id: 19,
+          body: "always happy",
+          article_id: 1,
+          author: "rogersop",
+          votes: 0,
+          created_at: expect.any(String),
+        });
+      });
+  });
+
+  test("Post: 422 adds a new comment to the db where the foreign key doesnt exist", () => {
+    const newComment = {
+      username: "doesnt exist",
+      body: "always happy",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(422)
+      .then((response) => {
+        expect(response.body.msg).toBe("Unprocessable Entity");
+      });
+  });
+
+  test("Post: 422 sends an appropriate status error message when given a valid post but non-existent id", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "always happy",
+    };
+    return request(app)
+      .post("/api/articles/399/comments")
+      .send(newComment)
+      .expect(422)
+      .then((response) => {
+        expect(response.body.msg).toBe("Unprocessable Entity");
```

Comment:
Again probably a 404 not found.

Reply:
Very nice work. I would also add a 400 /api/articles/notanumber/comments test :)